web/src/components/vis/VolcanoPlot.vue

```diff
@@ -23,106 +46,160 @@ export default {
+    });
+
+    const transformedRows = computed(() => props.rows.map((row) => ({
+      ...row,
+      x: row.log2FoldChange,
+      y: -Math.log10(row.pValue),
+    }))) as Ref<TransformedRow[]>;
+
+    const xrange = computed(() => {
+      const max = transformedRows.value
+        .reduce((acc: number, d: TransformedRow) => Math.max(acc, Math.abs(d.x)), 0);
       return [-max, max];
-    },
-    yrange() {
-      const max = this.transformedRows.reduce((acc, d) => Math.max(acc, d.y), 0);
+    }) as Ref<[number, number]>;
+
+    const yrange = computed(() => {
+      const max = transformedRows.value
+        .reduce((acc: number, d: TransformedRow) => Math.max(acc, d.y), 0);
       return [0, max];
-    },
-  },
-  watch: {
-    rows() {
-      this.update();
-    },
-    minFoldChange() {
-      this.update();
-    },
-    minLogP() {
-      this.update();
-    },
-  },
-  mounted() {
-    this.update();
-  },
-  methods: {
-    update() {
-      // Compute the total variance in all the PCs.
-      const svg = select(this.$refs.svg);
-      this.axisPlot(svg);
+    }) as Ref<[number, number]>;
+
+    const scaleX = computed(() => scaleLinear()
+      .domain(xrange.value)
+      .range([0, dwidth.value]));
+
+    const scaleY = computed(() => scaleLinear()
+      // @ts-ignore d3 typings are bad
+      .domain(yrange.value)
+      // @ts-ignore d3 typings are bad
+      .range([dheight.value, 0]));
+
+    const axisX = computed(() => axisBottom(scaleX.value));
+    const axisY = computed(() => axisLeft(scaleY.value));
+
+    function onResize() {
+      const bb = mainRef.value.getBoundingClientRect();
+      width.value = bb.width;
+    }
+    onMounted(() => onResize());
+    watchEffect(() => {
+      const svg = select('svg');
+      axisPlot(svg, axisX.value, axisY.value);
 
+      const _scaleX = scaleX.value;
+      const _scaleY = scaleY.value;
+
+      // @ts-ignore
       svg.select('.plot').selectAll('circle')
-        .data(this.transformedRows)
+        .data(transformedRows.value)
         .join((enter) => {
           const r = enter.append('circle');
           r.append('title');
           return r;
         })
-        .attr('r', (d) => (Math.abs(d.x) >= this.minFoldChange && d.y >= this.minLogP ? this.radius * 2 : this.radius))
-        .attr('opacity', (d) => (Math.abs(d.x) >= this.minFoldChange && d.y >= this.minLogP ? 1 : 0.5))
-        .attr('cx', (d) => this.scaleX(d.x))
-        .attr('cy', (d) => this.scaleY(d.y))
+        // @ts-ignore
+        .attr('r', (d) => (Math.abs(d.x) >= props.minFoldChange && d.y >= props.minLogP ? radius * 2 : radius))
+        // @ts-ignore
+        .attr('opacity', (d) => (Math.abs(d.x) >= props.minFoldChange && d.y >= props.minLogP ? 1 : 0.5))
+        // @ts-ignore
+        .attr('cx', (d) => _scaleX(d.x))
+        // @ts-ignore
+        .attr('cy', (d) => _scaleY(d.y))
+        // @ts-ignore
         .style('fill', (d) => d.color)
         .select('title')
+        // @ts-ignore
```

Comment:
This is a depressing number of `@ts-ignore`. Is there a way to resolve whatever it's whining about, or to ignore a whole block?

Reply:
It turns out only one `@ts-ignore` was needed at the start of the method chaining. I worked from the bottom of the file up when I was fixing those typescript errors so I missed that.